stdlib/public/core/Character.swift

```diff
@@ -125,11 +125,29 @@ public struct Character :
     utf8CodeUnitCount: Builtin.Word,
     isASCII: Builtin.Int1
   ) {
-    self = Character(
-      String(
-        _builtinExtendedGraphemeClusterLiteral: start,
-        utf8CodeUnitCount: utf8CodeUnitCount,
-        isASCII: isASCII))
+    // Most character literals are going to be fewer than eight UTF-8 code
+    // units; for those, build the small character representation directly.
+    if _fastPath(Int(utf8CodeUnitCount) <= 8) {
+      var buffer: UInt64 = ~0
+      _memcpy(
+        dest: UnsafeMutableRawPointer(Builtin.addressof(&buffer)),
+        src: UnsafeMutableRawPointer(start),
+        size: UInt(utf8CodeUnitCount))
+      let utf8Chunk = UInt64(littleEndian: buffer)
```

Comment:
The `.small`. representation orders the packed code units from low to high bits. Since we're doing a `memcpy` of code units directly into the memory occupied by a `UInt64`, it's assuming little-endianness everywhere, and the `UInt64.init(littleEndian:)` call is necessary to ensure the correct integer representation on big-endian systems. The same logic—which I adapted—already exists in the function called by `Character.init(String:)` [here](https://github.com/apple/swift/blob/6d1ae2a39c1b77240107854b0ae1a35800a8ba73/stdlib/public/core/StringUTF8.swift#L52).